pkg/runctl/controller.go

```diff
@@ -246,25 +246,24 @@ func (c *Controller) syncHandler(key string) error {
 	runManager := c.createRunManager(pipelineRun, pipelineRunsConfig)
 	// Process pipeline run based on current state
 	switch state := pipelineRun.GetStatus().State; state {
-	// TODO fix #117
-	// Runs might be left in state `preparing` after a controller crash.
-	// Those must be recovered.
 	case api.StateUndefined:
-		c.metrics.CountStart()
 		if err = c.changeState(pipelineRun, api.StatePreparing); err != nil {
 			return err
 		}
+		c.metrics.CountStart()
+	case api.StatePreparing:
 		err = runManager.Start(pipelineRun)
 		if err != nil {
+			// create Event here
 			pipelineRun.StoreErrorAsMessage(err, "error syncing resource")
-			if pipelineRun.GetStatus().Result == api.ResultUndefined {
-				pipelineRun.UpdateResult(api.ResultErrorInfra)
-			}
-			if err = c.changeState(pipelineRun, api.StateCleaning); err != nil {
-				return err
+			if pipelineRun.GetStatus().Result != api.ResultUndefined {
```

Comment:
```suggestion
			//In case we have a result we can cleanup. Otherwise we retry in the next iteration.
			if pipelineRun.GetStatus().Result != api.ResultUndefined {
```